### DISCUSSION TITLE
Twitter links are not working (t.co) despite whitelistet

### DISCUSSION THREAD

**Discussion by Raptor05** [AdguardTeam/AdGuardHome#6013] [Q&A] [answered]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Custom (please mention in the description)

### Installation

Docker

### Setup

On one machine

### AdGuard Home version

v0.107.33

### Action

Open a link in twitter. Twitter is using an URL shortener (t.co). 

### Expected result

The twitter shortened link opens.

### Actual result

I get 'ERR_ADDRESS_INVALID' error whenever I'm in my internal network, with each device.

### Additional information and/or screenshots

- AdGuard Home protocol shows that t.co is whitelisted (My used [List](https://raw.githubusercontent.com/hg1978/AdGuard-Home-Whitelist/master/whitelist.txt))
- Even when I deactivate AdGuard completely, the URL is not reachable
- When I'm using mobile network on mobile phone, URL's are working
- When I'm changing my DNS server in device, URL's are working
- I'm using AdGuard in docker (Synology DS918+)
- My router is a AVM FritzBox 7490
- I'm using AdGuard as DNS server for the Router itself + as advertisement for each device (IP4 and IP6)

```sh
nslookup -debug -type=a 'https://t.co/' '$YOUR_AGH_ADDRESS'
```
```
*** Die Serveradresse für "'$YOUR_AGH_ADDRESS'" kann nicht gefunden werden:
------------
Got answer:
    HEADER:
        opcode = QUERY, id = 1, rcode = NOERROR
        header flags:  response, auth. answer, want recursion, recursion avail.
        questions = 1,  answers = 1,  authority records = 1,  additional = 3

    QUESTIONS:
        2.178.168.192.in-addr.arpa, type = PTR, class = IN
    ANSWERS:
    ->  2.178.168.192.in-addr.arpa
        name = 03de06d**hidden**8d321735f4adaf02.fritz.box
        ttl = 9 (9 secs)
    AUTHORITY RECORDS:
    ->  2.178.168.192.in-addr.arpa
        nameserver = fritz.box
        ttl = 9 (9 secs)
    ADDITIONAL RECORDS:
    ->  fritz.box
        internet address = 192.168.178.1
        ttl = 9 (9 secs)
    ->  fritz.box
        AAAA IPv6 address = fd44:**hidden**:9685
        ttl = 9 (9 secs)
    ->  fritz.box
        AAAA IPv6 address = 2001:9e8:**hidden**:d5ff:fe66:9685
        ttl = 9 (9 secs)

------------
Server:  03de06d02e1**hidden**1735f4adaf02.fritz.box
Address:  192.168.178.2

------------
Got answer:
    HEADER:
        opcode = QUERY, id = 2, rcode = NXDOMAIN
        header flags:  response, want recursion, recursion avail.
        questions = 1,  answers = 0,  authority records = 1,  additional = 0

    QUESTIONS:
        'https://t.co/'.**hidden**.net, type = A, class = IN
    AUTHORITY RECORDS:
    ->  **hidden**.net
        ttl = 163 (2 mins 43 secs)
        primary name server = ns1-03.azure-dns.com
        responsible mail addr = azuredns-hostmaster.microsoft.com
        serial  = 1
        refresh = 3600 (1 hour)
        retry   = 300 (5 mins)
        expire  = 2419200 (28 days)
        default TTL = 300 (5 mins)

------------
------------
Got answer:
    HEADER:
        opcode = QUERY, id = 3, rcode = NXDOMAIN
        header flags:  response, want recursion, recursion avail.
        questions = 1,  answers = 0,  authority records = 1,  additional = 0

    QUESTIONS:
        'https://t.co/', type = A, class = IN
    AUTHORITY RECORDS:
    ->  (root)
        ttl = 833 (13 mins 53 secs)
        primary name server = a.root-servers.net
        responsible mail addr = nstld.verisign-grs.com
        serial  = 2023071400
        refresh = 1800 (30 mins)
        retry   = 900 (15 mins)
        expire  = 604800 (7 days)
        default TTL = 86400 (1 day)

------------
*** 'https://t.co/' wurde von 03de06**hidden**21735f4adaf02.fritz.box nicht gefunden: Non-existent domain.
```

--> I'm not sure what to add for "$YOUR_AGH_ADDRESS"

**Comment by ainar-g** [comment]
Hello.  Your `nslookup` command is incorrect, since DNS works on the domain-name level, not on the URL level.  So, assuming that the address of AdGuard Home is `192.168.1.1`:

```sh
nslookup -debug -type=a 't.co' 192.168.1.1
```

(Replace the IP address if it's not.)

Also, what do you see in the Query log whenever you do a request for `t.co`?  It could also be blocked by a CNAME or an IP address, depending on the response from the upstream servers.

**Reply by Raptor05** [reply]
Hey,
thanks for your response.
My AdGuard Home is running on 192.168.178.2. But with that IP, the nslookup is failing:
```
\>nslookup -debug -type=a 't.co' 192.168.172.2
DNS request timed out.
    timeout was 2 seconds.
timeout (2 secs)
Server:  UnKnown
Address:  192.168.172.2

DNS request timed out.
    timeout was 2 seconds.
timeout (2 secs)
DNS request timed out.
    timeout was 2 seconds.
timeout (2 secs)
*** Zeitüberschreitung bei Anforderung an UnKnown.
```

The query log looks a little bit strange, isn't it?
```
Response details
Status: Authorized
DNS-Server: tls://fdns1.dismail.de:853
Elapsed: 21 ms
Returncode: NOERROR
Rule(s): @@||t.co^$important Default Whitelist
Response: A: 0.0.0.0 (ttl=10)
```
(I've translated the labels)

**Reply by Raptor05** [reply, marked as answer]
I've removed the tls://fdns1.dismail.de DNS server from my AdGuard Config and now the twitter links are working again.

It seems that that DNS server is blocking t.co URLs.

**Comment by ppfeufer** [comment]
![image](https://github.com/AdguardTeam/AdGuardHome/assets/2989985/3b5c9899-15fa-4929-8bcc-291ac9aa8e53)
Yep, that DNS server blocks stuff through included block lists, see here » https://dismail.de/info.html#dns

**Comment by lastmetroid** [comment]
I have the same issue, how do I make it work? I use these filter lists

![image](https://github.com/user-attachments/assets/d2c90c0c-8d72-459f-8ac1-6055bd01d433)

![image](https://github.com/user-attachments/assets/9a9f372e-257e-4ef4-b1cf-c2214bf7f53b)

How to add white list entry for t.co ? Would this work?

![image](https://github.com/user-attachments/assets/57ff4be4-94e3-42fd-956e-f3f8cc06c455)

Update:

Ok this seems to work
